### PR TITLE
[C11116] Unlinking a match or action profile from a job profile (folijet) (TaaS)

### PR DESCRIPTION
**Pull request by AliaksandrShuleika**
[testrail](https://foliotest.testrail.io/index.php?/cases/view/11116)
[report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4922/allure/#suites/0f29a4212ddbc0a40cf0646f9bcb6758/14bebca0d448451a/)
image

### CODE REVIEW

**Comment by manowarqqq** [comment]
Please, replace allure report with valid link and add screenshot from this report because this link will be expired soon

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        ActionProfileView.close();
+        ActionProfiles.waitLoading();
+        ActionProfiles.create(actionProfile2, mappingProfile.name);
+        ActionProfileView.close();
+        ActionProfiles.waitLoading();
+        ActionProfiles.create(actionProfile3, mappingProfile.name);
+        ActionProfileView.close();
+        ActionProfiles.waitLoading();
+
+        // create match profile
```
Comment (manowarqqq) [inline cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:89]: could be  done with  NewMatchProfile.createMatchProfileViaApi

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      profileName: `C11116 autotest job profile ${getRandomPostfix()}`,
+      acceptedType: ACCEPTED_DATA_TYPE_NAMES.MARC,
+    };
+
+    before('create test data', () => {
+      cy.createTempUser([Permissions.settingsDataImportEnabled.gui]).then((userProperties) => {
+        user = userProperties;
+        cy.login(userProperties.username, userProperties.password);
+
+        // create mapping profile
```
Comment (manowarqqq) [inline cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:68]: could be done with NewFieldMappingProfile.createMappingProfileViaApi

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      MatchProfiles.deleteMatchProfile(matchProfile.profileName);
+      Users.deleteViaApi(user.userId);
+    });
+
+    it(
+      'C11116 Unlinking a match or action profile from a job profile (folijet) (TaaS)',
+      { tags: [TestTypes.extendedPath, DevTeams.folijet] },
+      () => {
+        // create Job profile with linked match and action profiles
+        cy.visit(SettingsMenu.jobProfilePath);
```
Comment (manowarqqq) [inline cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:110]: First step is open Action Profiles, not Job Profiles

File: cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js [hunk 4]
```diff
@@ -17,4 +17,15 @@ export default {
       .click();
     cy.do(Modal({ id: 'unlink-job-profile-modal' }).find(Button('Unlink')).click());
   },
+  unlinkMatchProfile: (number) => {
+    cy.get('[id*="branch-ROOT-MATCH-editable"]').eq(number).find('button[icon="unlink"]').click();
```
Comment (manowarqqq) [inline cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js:21]: Please use Interactors

File: cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js [hunk 5]
```diff
@@ -17,4 +17,15 @@ export default {
...
+    cy.do(Modal({ id: 'unlink-job-profile-modal' }).find(Button('Unlink')).click());
+  },
+  unlinkNonMatchProfile: (number) => {
+    cy.get('[id*="branch-ROOT-NON_MATCH-editable"]')
```
Comment (manowarqqq) [inline cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js:25]: Please use Interactors

File: cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js (hunk 4 above)
Comment (AliaksandrShuleika) [reply cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js:21]: what interactor i can use for this element [id*="branch-ROOT-MATCH-editable"]?

File: cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js (hunk 5 above)
Comment (AliaksandrShuleika) [reply cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js:25]: what interactor i can use for this element [id*="branch-ROOT-MATCH-editable"]?

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        // create mapping profile
+        cy.visit(SettingsMenu.mappingProfilePath);
+        FieldMappingProfiles.openNewMappingProfileForm();
+        NewFieldMappingProfile.fillSummaryInMappingProfile(mappingProfile);
+        NewFieldMappingProfile.fillInstanceStatusTerm(instanceStatusTerm);
+        NewFieldMappingProfile.fillCatalogedDate(catalogedDate);
+        NewFieldMappingProfile.save();
+        FieldMappingProfileView.closeViewMode(mappingProfile.name);
+
+        // create 3 action profiles
```
[the remark this answers is not in the document]
Comment (AliaksandrShuleika) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js]: fixed

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js (hunk 2 above)
Comment (AliaksandrShuleika) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:51]: fixed

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js (hunk 3 above)
Comment (AliaksandrShuleika) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:85]: fixed

File: cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js (hunk 4 above)
Comment (manowarqqq) [reply cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js:21]: There is no Interactor for such element for a while. But for button use Button interactor (find('button[icon="unlink"])

Comment (AliaksandrShuleika) [reply cypress/support/fragments/data_import/job_profiles/jobProfileEdit.js:21]: can you use interactors in cy.get statement?

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+import MatchProfiles from '../../../support/fragments/data_import/match_profiles/matchProfiles';
+import NewMatchProfile from '../../../support/fragments/data_import/match_profiles/newMatchProfile';
+import { ACCEPTED_DATA_TYPE_NAMES, FOLIO_RECORD_TYPE } from '../../../support/constants';
+import Users from '../../../support/fragments/users/users';
+import ActionProfileView from '../../../support/fragments/data_import/action_profiles/actionProfileView';
+
+describe('data-import', () => {
+  describe('Settings', () => {
+    let user;
+    const actionProfile1 = {
```
Comment (TetianaParanich) [inline cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:20]: You can create an array of objects and pit in it mapping and action profiles.

Comment (AliaksandrShuleika) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:20]: what array structure i can use when i need 3 action profiles 1 match profile 1 mapping profile and 1 job profile?

Comment (TetianaParanich) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:20]: I think you can try something like
const collectionOfActionProfiles = [
      { actionProfile: {
        typeValue: FOLIO_RECORD_TYPE.INSTANCE,
        name: `C11116 autotest action profile1 ${getRandomPostfix()}`,
      } },
      { actionProfile: {
        typeValue: FOLIO_RECORD_TYPE.INSTANCE,
        name: `C11116 autotest action profile2 ${getRandomPostfix()}`,
      } },
      { actionProfile: {
        typeValue: FOLIO_RECORD_TYPE.INSTANCE,
        name: `C11116 autotest action profile3 ${getRandomPostfix()}`,
      } },
    ];
    const matchProfile = {
      profileName: `C11116 autotest match profile ${getRandomPostfix()}`,
    };
    const mappingProfile = {
      name: `C11116 mapping profile ${getRandomPostfix()}`,
    };
    const jobProfile = {
      profileName: `C11116 autotest job profile ${getRandomPostfix()}`,
      acceptedType: ACCEPTED_DATA_TYPE_NAMES.MARC,
    };

    cy.visit(SettingsMenu.actionProfilePath);
    collectionOfActionProfiles.forEach((profile) => {
      ActionProfiles.create(profile.actionProfile, mappingProfile.name);
      ActionProfiles.checkActionProfilePresented(profile.actionProfile.name);
    });

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+      });
+    });
+
+    after('delete test data', () => {
+      JobProfiles.deleteJobProfile(jobProfile.profileName);
+      ActionProfiles.deleteActionProfile(actionProfile1.name);
+      ActionProfiles.deleteActionProfile(actionProfile2.name);
+      ActionProfiles.deleteActionProfile(actionProfile3.name);
+      FieldMappingProfileView.deleteViaApi(mappingProfile.name);
+      MatchProfiles.deleteMatchProfile(matchProfile.profileName);
```
Comment (TetianaParanich) [inline cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:65]: You should delete match profile after job profile and then only delete action profiles

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        NewFieldMappingProfile.createMappingProfileViaApi(mappingProfile.name);
+
+        // create match profile
+        NewMatchProfile.createMatchProfileViaApi(matchProfile.profileName);
+      });
+    });
+
+    after('delete test data', () => {
+      JobProfiles.deleteJobProfile(jobProfile.profileName);
+      ActionProfiles.deleteActionProfile(actionProfile1.name);
```
Comment (TetianaParanich) [inline cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:61]: You can use array for deleting
collectionOfActionProfiles.forEach((profile) => {
        ActionProfiles.deleteActionProfile(profile.actionProfile.name);
      });

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js (hunk 7 above)
Comment (AliaksandrShuleika) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js]: thanks. fixed

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js (hunk 9 above)
Comment (AliaksandrShuleika) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js]: fixed

File: cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js (hunk 8 above)
Comment (AliaksandrShuleika) [reply cypress/e2e/data-import/settings/unlink-from-job-profile.cy.js:63]: fixed

**Review by TetianaParanich — APPROVED** [review]

**Pull request merged**